Userland/Utilities/realpath.cpp

```diff
@@ -5,23 +5,22 @@
  */
 
 #include <LibCore/ArgsParser.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
 #include <stdio.h>
 #include <unistd.h>
 
-int main(int argc, char** argv)
+ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
-    if (pledge("stdio rpath", nullptr) < 0) {
-        perror("pledge");
-        return 1;
-    }
+    TRY(Core::System::pledge("stdio rpath"));
 
     const char* path;
 
     Core::ArgsParser args_parser;
     args_parser.set_general_help(
         "Show the 'real' path of a file, by resolving all symbolic links along the way.");
     args_parser.add_positional_argument(path, "Path to resolve", "path");
-    args_parser.parse(argc, argv);
+    args_parser.parse(arguments);
 
     char* value = realpath(path, nullptr);
```

Comment:
This could have a wrapper added to `LibCore/System.[h,cpp]`, to make the API nicer. (Return `ErrorOr<String>` instead or a raw pointer, etc)